### PR TITLE
Add automatic package onboarding

### PR DESCRIPTION
**Pull request by danieljurek**
Add automatic package onboarding and basic verification for changed/new packages

Example successful pipeline execution: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1047876&view=logs&j=dc056dfc-c0cf-5958-c8c4-8da4f91a3739&t=51aa7976-5924-5040-41de-f124a4debc20

Example delta from execution: https://github.com/Azure/azure-docs-sdk-java/commit/e87d08c5ef710bbf0347cba9bd37411e98c1dae0

### CODE REVIEW

File: eng/scripts/Language-Settings.ps1 [hunk 1]
```diff
@@ -260,6 +260,260 @@ function Update-java-CIConfig($pkgs, $ciRepo, $locationInDocRepo, $monikerId=$nu
...
+  UpdateDocsMsPackages `
+    (Join-Path $DocsRepoLocation 'package.json') `
+    'latest' `
+    $FilteredMetadata
+}
+
+function UpdateDocsMsPackages($DocConfigFile, $Mode, $DocsMetadata) {
+  $packageConfig = Get-Content $DocConfigFile -Raw | ConvertFrom-Json
+
+  $workingDirectory = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:376]: Why do we need a temp directory? If this is to extract and parse the package do we think we can switch to using the package properties instead?

Comment (weshaggard) [reply eng/scripts/Language-Settings.ps1:376]: Or is this more for checking the dependencies?

File: eng/scripts/Language-Settings.ps1 [hunk 2]
```diff
@@ -260,6 +260,260 @@ function Update-java-CIConfig($pkgs, $ciRepo, $locationInDocRepo, $monikerId=$nu
...
+}
+
+function PackageDependenciesResolve($package, $workingDirectory) { 
+  $packageDirectory = Join-Path `
+    $workingDirectory `
+    "$($package.packageGroupId)__$($package.packageArtifactId)__$($package.packageVersion)"
+  New-Item -ItemType Directory -Path $packageDirectory -Force
+  
+  $pomArtifactName = "$($package.packageGroupId):$($package.packageArtifactId):$($package.packageVersion):pom"
+  $artifactDownloadOutput = mvn `
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:318]: Is this duplicated between here and SourcePackageHasComFolder? Should we only be running it once or is there a difference I cannot easily see?

File: eng/scripts/Language-Settings.ps1 (hunk 1 above)
Comment (danieljurek) [reply eng/scripts/Language-Settings.ps1:376]: This is for the purpose of validating that the package will pass the docs CI (dependency checking, having a `com/` folder in the source jar). 

We could easily create a temp directory for every package we validate but that could get difficult to track down when running locally.

File: eng/scripts/Language-Settings.ps1 (hunk 2 above)
Comment (danieljurek) [reply eng/scripts/Language-Settings.ps1:318]: The difference is at the end: 

Download the sources jar (`...:jar:sources`) -- 
```
"$($package.packageGroupId):$($package.packageArtifactId):$($package.packageVersion):jar:sources"
```

Download the pom (`...:pom`)  --
```
"$($package.packageGroupId):$($package.packageArtifactId):$($package.packageVersion):pom"
```

However the leading portion could be created in a higher level function as well as the package-specific temp directory.

File: eng/scripts/Language-Settings.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/scripts/Language-Settings.ps1:376]: Ah it isn't about getting simple properties. I agree with you that having one temp directory is better then having a unique one for each package.

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/docindex.yml [hunk 3]
```diff
@@ -2,3 +2,44 @@ trigger: none
 
 jobs:
   - template: /eng/common/pipelines/templates/jobs/docindex.yml
+
```
Comment (benbp) [inline eng/pipelines/docindex.yml:5]: Is this eventually going to go into tools/common?

File: eng/pipelines/docindex.yml [hunk 4]
```diff
@@ -2,3 +2,44 @@ trigger: none
...
+  - job: UpdateDocsMsBuildConfig
+    pool:
+      vmImage: ubuntu-20.04
```
Comment (benbp) [inline eng/pipelines/docindex.yml:8]: Do you not want to use the 1es pool?

File: eng/pipelines/docindex.yml (hunk 3 above)
Comment (danieljurek) [reply eng/pipelines/docindex.yml:5]: These changes are written to be minimally differentiated and easy to put into a template. We should always have a platform specific `docindex.yml` file so we can opt repos into or out of docs.ms publishing processes.

File: eng/pipelines/docindex.yml (hunk 4 above)
Comment (danieljurek) [reply eng/pipelines/docindex.yml:8]: We could. I have no strong opinions. 

Arguments for using the hosted pool for engsys tools as opposed to the 1ES pools: 
* 1ES pools may saturate from time to time and using a different pool gives us the ability to run some processes without having to wait for machines to become available 
* The capacity in these pools can be used at comparatively lower cost

**Pull request merged**